File: app/src/main/java/com/konifar/materialcat/views/ListLoadingView.java
package com.konifar.materialcat.views;

import android.content.Context;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.LinearLayout;

import com.konifar.materialcat.R;

import butterknife.Bind;
import butterknife.ButterKnife;

public class ListLoadingView extends LinearLayout {

    @Bind(R.id.list_loading)
    FrameLayout mListLoading;

    public ListLoadingView(Context context) {
        super(context);
        inflate(context, R.layout.ui_list_loading, this);
        ButterKnife.bind(this);
    }

    public void switchVisible(boolean visible) {
        int visibility = visible ? View.VISIBLE : View.GONE;
        mListLoading.setVisibility(visibility);
    }

}

File: app/src/main/kotlin/com/konifar/materialcat/views/ListLoadingView.kt
package com.konifar.materialcat.views

import android.content.Context
import android.databinding.DataBindingUtil
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import com.konifar.materialcat.R
import com.konifar.materialcat.databinding.UiListLoadingBinding

class ListLoadingView(context: Context) : LinearLayout(context) {

    var binding: UiListLoadingBinding

    init {
        binding = DataBindingUtil.inflate<UiListLoadingBinding>(LayoutInflater.from(context), R.layout.ui_list_loading, this, true)
    }

    fun switchVisible(visible: Boolean) {
        val visibility = if (visible) View.VISIBLE else View.GONE
        binding.listLoading.visibility = visibility
    }

}
